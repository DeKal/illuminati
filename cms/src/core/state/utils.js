export const toSuccess = actionName => `${actionName}Success`

export const toError = actionName => `${actionName}Error`

export const toRequest = actionName => `${actionName}Request`
