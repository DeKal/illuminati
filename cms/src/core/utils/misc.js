import { POSTS_PER_PAGE } from 'core/const/posts'

export const isPathMatchUrl = (pathName, url, isExact = false) => {
  if (url === '/' || pathName === '/' || isExact) {
    return pathName === url
  }
  return pathName.includes(url)
}

export const stringtifyTags = (tags, limits = 20) => {
  const tagsStr = tags.join(', ')
  if (tagsStr.length > limits) {
    return tagsStr.slice(0, limits) + '...'
  }
  return tagsStr
}

export const calcMaxRowsPerPage = postCount => {
  const postsPerPageOptions = []
  for (let index = 0; index < POSTS_PER_PAGE.length; index++) {
    postsPerPageOptions.push(POSTS_PER_PAGE[index])
    if (POSTS_PER_PAGE[index] > postCount) {
      break
    }
  }
  return postsPerPageOptions
}

export const arrayToAttributes = array =>
  array.reduce((object, attribute) => {
    return {
      ...object,
      [attribute]: true
    }
  }, {})

export const attributesToArray = obj =>
  Object.keys(obj).map(attribute => (obj[attribute] ? attribute : null))
