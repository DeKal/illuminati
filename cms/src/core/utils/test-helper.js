import React from 'react'
import { shallow, mount } from 'enzyme'
import Theme from 'core/theme/ThemeWrapper'

export const shallowWithTheme = component => {
  const wrapper = shallow(
    <Theme>
      <div data-test-id="theme-wrapper">{component}</div>
    </Theme>
  )
  return wrapper.find('[data-test-id="theme-wrapper"]').shallow()
}

export const mountWithTheme = component => {
  const wrapper = mount(
    <Theme>
      <div data-test-id="theme-wrapper">{component}</div>
    </Theme>
  )
  return wrapper.find('[data-test-id="theme-wrapper"]')
}
