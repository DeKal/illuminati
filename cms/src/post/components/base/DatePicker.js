import { styled } from '@material-ui/core/styles'
import { KeyboardDatePicker } from '@material-ui/pickers'

const DatePicker = styled(KeyboardDatePicker)(({ theme }) => ({
  padding: theme.spacing(1),
  margin: 0,
  '& .MuiInputLabel': {
    '&-formControl': {
      padding: theme.spacing(1),
      transform: 'translate(0, 1.5px) scale(0.75)'
    }
  },
  '& .MuiInput': {
    '&-underline:after': {
      borderBottom: `2px solid ${theme.colors.primary}`
    }
  }
}))

export default DatePicker
