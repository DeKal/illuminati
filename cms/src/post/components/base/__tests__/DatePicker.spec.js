import * as React from 'react'
import { shallowWithTheme } from 'core/utils/test-helper'
import DatePicker from 'post/components/base/DatePicker'

describe('DatePicker', () => {
  it('should render correctly', function() {
    const wrap = shallowWithTheme(<DatePicker />)
    expect(wrap).toMatchSnapshot()
  })
})
