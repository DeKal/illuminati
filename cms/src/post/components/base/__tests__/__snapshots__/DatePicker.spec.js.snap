// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker should render correctly 1`] = `
<div
  data-test-id="theme-wrapper"
>
  <Styled(PickerWithState) />
</div>
`;
